ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -335,11 +338,19 @@ static boolean registerMBean(
   }
 
   /**
-   * The peer belongs to the current configuration, should start as a follower
+   * The peer belongs to the current configuration, should start as a follower or listener
    */
-  private void startAsFollower() {
-    setRole(RaftPeerRole.FOLLOWER, "startAsFollower");
-    role.startFollowerState(this, "startAsFollower");
+  private void startAsPeer(RaftPeerRole roles) {
+    Object reason = "";
+    if (roles.equals(RaftPeerRole.FOLLOWER)) {
+      reason = "startAsFollower";
+      setRole(RaftPeerRole.FOLLOWER, reason);
+    } else if (roles.equals(RaftPeerRole.LISTENER)) {
```

Comment:
Since RaftPeerRole is enum, use == .  Also, let's add an else-clause to throw an exception.
```
    if (newRole == RaftPeerRole.FOLLOWER) {
      reason = "startAsFollower";
      setRole(RaftPeerRole.FOLLOWER, reason);
    } else if (newRole == RaftPeerRole.LISTENER) {
      reason = "startAsListener";
      setRole(RaftPeerRole.LISTENER, reason);
    } else {
      throw new IllegalArgumentException("Unexpected role " + newRole);
    }
```

Reply:
done